x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Top.java

```diff
@@ -388,7 +388,8 @@ public Expression surrogate() {
         if (outputField() != null && field().semanticEquals(outputField())) {
             return new Top(s, field(), limitField(), orderField(), null);
         }
-        if (orderField() instanceof Literal && limitField() instanceof Literal && limitValue() == 1) {
+        // To replace Top by Min or Max, we cannot have an `outputField`
+        if (orderField() instanceof Literal && limitField() instanceof Literal && limitValue() == 1 && outputField() == null) {
```

Comment:
Can we add some CSV test reproducing this case? To ensure that this indeed failed before the fix.
A surrogate changing the result type of the expression is a quite hardcore error at planning time.
Our unit tests happen to execute the surrogate logic, but it's really executed as part of the planning, so having an integration test checking it looks safer (Apart of the unit tests, which look fine)

Reply:
Done. If I backtrack the fix, the following error is thrown in those tests (which I find not very helpful), so you are right the planner would not let this through:

```
Output has changed from [[shortest_employee{r}#577, tallest_employee{r}#581]] to [[shortest_employee{r}#577, tallest_employee{r}#581]].
```